pkg/model/api/v1/common/url.go

```diff
@@ -48,6 +48,9 @@ func NewURL(u *URL, paths ...string) *URL {
 }
 
 type URL struct {
```

Comment:
these annotations should go on the above type.

```suggestion
// +kubebuilder:validation:Schemaless
// +kubebuilder:validation:Type=string
// +kubebuilder:validation:Format=uri
type URL struct {
	*url.URL `json:"-" yaml:"-"`
}